build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -12,12 +12,66 @@ import { PackageJson } from "../common/npmPackage";
 import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
 import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
 import { typeOnly } from "./compatibility";
+import { ExtractorConfig } from "@microsoft/api-extractor";
 
 // Do not check that file exists before opening:
 // Doing so is a time of use vs time of check issue so opening the file could fail anyway.
 // Do not catch error from opening file since the default behavior is fine (exits process with error showing useful message)
 const packageObject: PackageJson = readJsonSync("package.json");
+const previousPackageName = `${packageObject.name}-previous`;
+
+const previousBasePath = `./node_modules/${previousPackageName}`;
+
+const previousPackageJsonPath = `${previousBasePath}/package.json`;
+
+if (!existsSync(previousPackageJsonPath)) {
+	throw new Error(
+		`${previousPackageJsonPath} not found. You may need to install the package via pnpm install.`,
+	);
+}
+const typeRollupFilePath = ExtractorConfig.tryLoadForFolder({
+	startingFolder: previousBasePath,
+});
 
+let typeDefinitionFilePath;
+
+if (typeRollupFilePath) {
+	// Check if a specified typeRollupFile exists
+	if (typeRollupFilePath === undefined) {
+		throw new Error(`Type rollup file '${typeRollupFilePath}' not found.`);
+	}
+	typeDefinitionFilePath = typeRollupFilePath;
+} else {
+	// Try to read the 'types' field from the package.json of the previous package
```

Comment:
@tylerbutler is this code block more along the lines of what you were expecting in terms of reading from types/exports from the previous package?